src/main/java/net/revelc/code/impsort/EmptyFileException.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.revelc.code.impsort;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Signals that the file denoted by this path is empty.
+ *
+ * <p>
+ * This exception will be thrown by the {@link ImpSort#parseFile} when it encounters an empty file.
+ * </p>
+ */
+public class EmptyFileException extends IOException {
+  private static final long serialVersionUID = -4202864513494828247L;
+
+  private final Path path;
+
+  /**
+   * Constructs a {@code EmptyFileException} with {@code null} as its error detail message and the
+   * specified path.
+   *
+   * @param path the path
+   */
+  public EmptyFileException(final Path path) {
+    this.path = path;
+  }
```

Comment:
It would be better to call super() with a detail message that includes the path than to rely on the default constructor in the super class and a null message. It's more friendly to those debugging, especially if it shows up after a breakpoint in a debugger tool, or if it leaks out into a stack trace somewhere.

Reply:
So, you want to add a `String` parameter to that and do `super(String)`, right?

Reply:
Yes, please. :smiley_cat: